crates/noirc_evaluator/src/ssa/acir_gen/operations/binary.rs

```diff
@@ -145,17 +145,24 @@ pub(crate) fn evaluate(
                 let l_c = var_cache.get_or_compute_internal_var_unwrap(binary.lhs, evaluator, ctx);
                 let mut r_c = var_cache.get_or_compute_internal_var_unwrap(binary.rhs, evaluator, ctx);
                 let predicate = get_predicate(var_cache,binary, evaluator, ctx).expression().clone();
-                //TODO avoid creating witnesses here.
-                let x_witness = r_c.get_or_compute_witness(evaluator, false).expect("unexpected constant expression"); 
-
-                let inverse = Expression::from(&constraints::evaluate_inverse(
-                    x_witness, &predicate, evaluator,
-                ));
-                InternalVar::from(constraints::mul_with_witness(
-                    evaluator,
-                    l_c.expression(),
-                    &inverse,
-                ))
+                if let Some(r_value) = r_c.to_const() {
+                    if r_value.is_zero() {
+                        panic!("Panic - division by zero");
+                    } else {
+                        constraints::add(&Expression::zero(), r_value.inverse(), l_c.expression()).into()
```

Comment:
You don't need the else branch because if the panic happens, then execution stops

Reply:
is that an issue?

Reply:
It's a nit, but it's not blocking